src/cascadia/WindowsTerminal/TrayIcon.cpp

```diff
@@ -161,17 +161,21 @@ HMENU TrayIcon::_CreateTrayContextMenu(IMapView<uint64_t, winrt::hstring> peasan
         // Submenu for Windows
         if (auto submenu = CreatePopupMenu())
         {
-            const auto locWindow = RS_(L"WindowIdLabel");
-            const auto locUnnamed = RS_(L"UnnamedWindowName");
-            for (const auto [id, name] : peasants)
+            for (const auto& p : peasants)
             {
-                winrt::hstring displayText = name;
-                if (name.empty())
+                winrt::hstring displayText{ fmt::format(L"#{} : ", p.Id) };
+
+                if (!p.TabTitle.empty())
+                {
+                    displayText = fmt::format(L"{} {}", displayText, p.TabTitle);
+                }
+
+                if (!p.Name.empty())
                 {
-                    displayText = fmt::format(L"{} {} - <{}>", locWindow, id, locUnnamed);
+                    displayText = fmt::format(L"{} [{}]", displayText, p.Name);
                 }

```

Comment:
`fmt` has a large binary size overhead and this code looks like it's building a string incrementally.
So it sounds like good ol' `std::wstringstream` might be a better fit here.

Alternatively you can also calculate your paddings yourself...
```cpp
static constexpr auto choose = [](const auto& str, const std::wstring_view& padding) {
	return str.empty() ? {} : padding;
};

fmt::format(
	L"#{} {:^{}}{}{}{}",
	p.Id,
	p.TabTitle,
	p.TabTitle.empty() ? 0 : p.TabTitle.size() + 2,
	p.Name.empty() ? : L"" : L"[",
	p.Name,
	p.Name.empty() ? : L"" : L"]"
);
```

I'm not sure which of the two is better, but I believe the current code isn't using `fmt` as it's intended. A `std::wstringstream` might be easier to understand though...

Reply:
wstringstream sounded simple enough to me 👍